pennylane/ops/pulse/convenience_functions.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains convenience functions for pulse programming."""
+
+import numpy as np
+
+has_jax = True
+try:
+    import jax.numpy as jnp
+except ImportError:
+    has_jax = False
+
+
+def pwc(t):
+    """Create a function that is piecewise-constant in time, based on the params for a TDHamiltonian.
+
+    Args:
+            t(Union[float, tuple(float, float)]: the total duration as a float, or the start and end time as floats.
+            index(int): the index at which the relevant parameter array is located in the overall ``params`` variable
+
+    Returns:
+            func: a function that can be passed the full ``params`` variable and ``t``, and will return the
```

Comment:
I'd avoid using specific names for the arguments (`params`, `t`). We could say instead:
```suggestion
            func: a function that contains two arguments, one for the trainable parameters and one for time, and returns the
```

Reply:
The `and will return the corresponding constant` phrase is not very clear. Can you explain it a bit more?

Reply:
See what you think of it now, does it seem clearer or just longer?

Reply:
lgtm!